avro-builder/builder/src/main/java/com/linkedin/avroutil1/builder/SchemaBuilder.java

```diff
@@ -248,7 +248,9 @@ public static void main(String[] args) throws Exception {
     if (options.has(enableUtf8Encoding)) {
       String value = options.valueOf(enableUtf8Encoding);
       handleUtf8Encoding = Boolean.TRUE.equals(Boolean.parseBoolean(value));
-      handleUtf8EncodingInPutByIndex = handleUtf8Encoding;
+      if (methodStringRepresentation.equals(StringRepresentation.CharSeq) && stringRepresentation.equals(StringRepresentation.CharSeq)) {
```

Comment:
This will not work for the user silently. Lets throw an illegalArgumentException and add this requirement in the option description above too.